### PR TITLE
Review and adapt date & time components to the new theme

### PR DESCRIPTION
**Pull request by CarlosCortizasCT**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=QGee2hj)

#### Summary

Review and adapt date & time components to the new theme

## Description

After the changes we made to the TextInput component for the new theme, we need to review the other input components to check whether an unexpected side effect has been made.

Since we are doing this review in chunks, in this particular one I reviewed these components:
- [x] DateInput
- [x] DateRangeInput
- [x] DateTimeInput
- [x] TimeInput

Also, I took the opportunity to adjust some colors regarding the icons. These changes where requested by the UX team.

### CODE REVIEW

File: design-system/materials/internals/definition.yaml [hunk 1]
```diff
@@ -317,6 +317,8 @@ componentGroups:
     description: 'Select input elements'
   select-input-option:
     description: 'Select input option elements'
+  datetime-input-icon:
```
Comment (CarlosCortizasCT) [inline design-system/materials/internals/definition.yaml:320]: This component group is shared by several date & time inputs.

File: packages/calendar-utils/src/calendar-body/calendar-body.styles.ts [hunk 2]
```diff
@@ -64,31 +67,37 @@ const getIconFontColor = (props: TCalendarBody) => {
...
+      &:hover:not(:disabled)&:not(:read-only),
+      &:focus {
+        border-color: ${designTokens.borderColorForInputWhenFocused};
+      }
+    `,
+    !isNewTheme &&
+      css`
+        border-left: 1px solid ${getIconBorderColor(props, state)};
+      `,
+  ];
```
Comment (CarlosCortizasCT) [inline packages/calendar-utils/src/calendar-body/calendar-body.styles.ts:100]: Here I just extracted the left border of the icon so it's only rendered in the default theme.

File: packages/calendar-utils/src/calendar-body/calendar-body.styles.ts [hunk 3]
```diff
@@ -124,42 +133,73 @@ const getInputFontColor = (props: TCalendarBody) => {
...
-      props.hasWarning ||
-      props.isReadOnly ||
-      ((props.isOpen || state.isFocused) && !props.isReadOnly)
-        ? ''
-        : designTokens.borderColorForInputWhenFocused};
-    }
-  `;
+      &:hover:not(:focus) {
+        background-color: ${designTokens.backgroundColorForInputWhenHovered};
+      }
```
Comment (CarlosCortizasCT) [inline packages/calendar-utils/src/calendar-body/calendar-body.styles.ts:164]: We need this `hover` to be located before the `focus-within & hover` so we can overwrite it.

**Comment by FilPob** [comment]
Just one issue: I think in the active, error and warning  state the inputs have a border-width of 3px. Should be 2px, same as text input

Otherwise looking good.

**Review by emmenko — APPROVED** [review]
Thanks a lot 💯

**Comment by CarlosCortizasCT** [comment]
> Just one issue: I think in the active, error and warning state the inputs have a border-width of 3px. Should be 2px, same as text input
> 
> Otherwise looking good.

Ouch! Sorry about that.

I've fixed it in this commit: ce702856572e43973373ec5e23d5b18070f02789

Please have another look whenever you have time.

Thanks.

**Review by ddouglasz — APPROVED** [review]
Looks good! 🚀

**Review by FilPob — APPROVED** [review]

**Pull request merged**